### ISSUE TITLE
Symphony sends (useless) cookies when redirecting

### ISSUE THREAD

**Issue by michael-e** [symphonycms/symphony-2#2494]
If you use Symphony's `redirect` function (from `/symphony/lib/boot/func.utilities.php`) while a page is being built (e.g. in an extension that hooks into a frontend delegate), Symphony will issue a potentially useless `Set-Cookie` header.

This is due to the fact that `cleanup_session_cookies()` is called late in `index.php` (Sym 2.3) resp. the `symphony_launcher` function (Sym 2.6), immediately before sending the rendered output. When redirecting, this funtion won't be called at all, so the `Set-Cookie` header won't be cleaned up.

I can't think of any reason to send such a header (or any other custom page header) with a redirect, so my proposal is to add `header_remove()` to Symphony's redirect function, like so:

```php
function redirect ($url){
	// Just make sure.
	$url = str_replace('Location:', null, $url);

	if(headers_sent($filename, $line)){
		echo "<h1>Error: Cannot redirect to <a href=\"$url\">$url</a></h1><p>Output has already started in $filename on line $line</p>";
		exit;
	}

    header_remove();
	header('Expires: Mon, 12 Dec 1982 06:00:00 GMT');
	header('Last-Modified: ' . gmdate('D, d M Y H:i:s') . ' GMT');
	header('Cache-Control: no-cache, must-revalidate, max-age=0');
	header('Pragma: no-cache');
	header("Location: $url");
	exit;
}
```

Can someone think of any counter-argument? If not, I can send a pull request, of course.

**Comment by michael-e** [comment]
The code example above is from Symphony 2.3, but IMHO the issue and the idea are just the same with Symphony 2.6.

**Comment by michael-e** [comment]
Doesn't work. When redirecting, for example, upon successful login, the session cookie will get lost.

I'll try `cleanup_session_cookies()`.

**Comment by brendo** [comment]
Did `cleanup_session_cookies()` do the trick?

**Comment by michael-e** [comment]
Yes, that works fine. I haven't found any side effects.

**Comment by michael-e** [comment]
It you like, I can send a PR (to the 2.6.x branch).

**Comment by brendo** [comment]
That'd be helpful, thanks!

**Comment by michael-e** [comment]
Will do, in an hour or so. (AFK for a while).

**Comment by michael-e** [comment]
The PR has been merged into the 2.6.x branch. I am not sure if we should close the issue, however, as long as it's not in master.

**Comment by nitriques** [comment]
>  I am not sure if we should close the issue, however, as long as it's not in master.

I think it should stay open until it's in master. I've created a new label, `Fixed in 2.6.x` to classify those.
